packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -150,13 +162,43 @@ export class Table extends Construct {
         }
     }
 
-    public addPartitionKey(name: string, type: KeyAttributeType): this {
-        this.addKey(name, type, HASH_KEY_TYPE);
+    /**
+     * @deprecated Use addPartitionKey(props: AttributeProps) instead.
+     */
+    public addPartitionKey(name: string, type: KeyAttributeType): this;
```

Comment:
No overloads please (does not translate to every jsii language).

Now is the time to break this API, if we're going to do so.

Reply:
I do not have a clear picture on backward compatibility rules. That is why I overloaded the method.

I would prefer breaking it if this is allowed. If this PR is a right direction to support GSI and LSI, the indexes will give users different API experience when defining partition and sort keys. It means that we will have API inconsistency which is not ideal.

Reply:
We are pre-1.0, feel free to break. Just make sure you follow conventional-commits to indicate the break